internal/modules/defaultmodule/defaultmodule.go

```diff
@@ -0,0 +1,151 @@
+	}
+	definedSignals := defs.DefinedSignal(signalInfo)
+	signalMap := make(map[string]*schema.SignalInfo, len(definedSignals))
+	for _, signal := range definedSignals {
+		signalMap[signal.JSONName] = signal
+	}
+
+	return &Module{signalMap: signalMap}, nil
+}
+
+// SetLogger sets the logger for the module.
+func (Module) SetLogger(*service.Logger) {}
+
+// SetConfig sets the configuration for the module.
+func (Module) SetConfig(string) error { return nil }
+
+// SignalConvert converts a default CloudEvent to DIMO's vss signals.
+func (m *Module) SignalConvert(_ context.Context, msgBytes []byte) ([]vss.Signal, error) {
+	signalEvent := cloudevent.CloudEvent[SignalData]{}
+	err := json.Unmarshal(msgBytes, &signalEvent)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal signal data: %w", err)
+	}
+	var decodeErrs error
+	vssSignals := make([]vss.Signal, 0)
+	for _, signal := range signalEvent.Data.Signals {
+		vssSig, err := m.defaultSignalToVSS(signal)
+		if err != nil {
+			// we want to return decoded signals even if some fail
+			decodeErrs = errors.Join(decodeErrs, err)
+			continue
+		}
+		vssSignals = append(vssSignals, vssSig)
+	}
+	return vssSignals, decodeErrs
+}
+
+func (m *Module) defaultSignalToVSS(signal *Signal) (vss.Signal, error) {
+	signalInfo, ok := m.signalMap[signal.Name]
+	if !ok {
+		return vss.Signal{}, fmt.Errorf("signal %s is not a defined signal name", signal.Name)
+	}
+	if signal.Value == nil {
+		return vss.Signal{}, fmt.Errorf("signal %s is missing a value", signal.Name)
+	}
+	vssSig := vss.Signal{
+		Timestamp: signal.Timestamp,
+		Name:      signal.Name,
+	}
+	switch signalInfo.BaseGoType {
+	case "float64":
+		num, ok := signal.Value.(float64)
+		if ok {
+			vssSig.ValueNumber = num
+		} else if str, ok := signal.Value.(string); ok {
+			v, err := strconv.ParseFloat(str, 64)
+			if err != nil {
+				return vss.Signal{}, fmt.Errorf("signal %s can not be converted to a float64: %w", signal.Name, err)
+			}
+			vssSig.ValueNumber = v
+		} else {
+			return vss.Signal{}, fmt.Errorf("signal %s is not a float64", signal.Name)
+		}
+	case "string":
+		str, ok := signal.Value.(string)
+		if !ok {
+			return vss.Signal{}, fmt.Errorf("signal %s is not a string", signal.Name)
+		}
+		vssSig.ValueString = str
+	default:
+		return vss.Signal{}, fmt.Errorf("signal %s has an unsupported base type %s", signal.Name, signalInfo.BaseGoType)
+	}
+
+	return vssSig, nil
+}
+
+// CloudEventConvert marshals the input message to Cloud Events and sets the type based on the message content.
+func (Module) CloudEventConvert(_ context.Context, msgData []byte) ([]cloudevent.CloudEventHeader, []byte, error) {
+	var event cloudevent.CloudEvent[json.RawMessage]
+	err := json.Unmarshal(msgData, &event)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to unmarshal message: %w", err)
+	}
+	hdrs := []cloudevent.CloudEventHeader{}
+	if gjson.GetBytes(event.Data, "vin").Exists() {
```

Comment:
Could we strip all the signals data from the fingerprint payload, do we need to store all of it or only vin portion?

Reply:
It doesn't create two files, only one.